### DISCUSSION TITLE
Failed to download mesh agent (Timeout)

### DISCUSSION THREAD

**Discussion by Aghiad90** [amidaware/tacticalrmm#1197] [Q&A]
Hello , this is new setup and i am trying to download and install the agent on a machine 
i tried it with -log debug and here is the output 
- No antivirus on the machine 
- i did add the exception on windows defender 
- i did disable windows defender 
- ports are open 
 it always stuck at this point 

![image](https://user-images.githubusercontent.com/23388761/177200864-306494b1-5653-4ee7-841f-5d8aec06fefb.png)

Regards ,

**Comment by dinger1986** [comment]
Is mesh.yourdomain.com working?

Are you running behind a proxy?

Do you have any agents installed?

**Reply by Aghiad90** [reply]
Hi , @dinger1986   
yes mesh.mydomain.com is working 
no proxy 
this is fresh install and i do not have any agent yet

**Reply by dinger1986** [reply]
Run the mesh troubleshooting command and share the output but hide the tokens etc

**Reply by Aghiad90** [reply]
@dinger1986  do you mind to share the command please ?

**Reply by dinger1986** [reply]
https://docs.tacticalrmm.com/install_agent/#stuck-at-downloading-mesh-agent

**Reply by Aghiad90** [reply]
@dinger1986 
![image](https://user-images.githubusercontent.com/23388761/177208881-b3699184-291c-4390-a5a3-df4b6842868f.png)

**Reply by dinger1986** [reply]
Hmm all looks fine.

Is there a firewall between the device and mesh? Can you download and install mesh directly?

**Reply by Aghiad90** [reply]
@dinger1986  
i did download it and install it manually and it did show up here 
![image](https://user-images.githubusercontent.com/23388761/177210245-01683358-3ab1-4f7a-98b8-f95250baa0f3.png)

but the device did not show up under TacticalRMM dashboard 
![image](https://user-images.githubusercontent.com/23388761/177210324-c7a6c8e8-33c0-4a86-993e-0dfe0744a8ce.png)

**Reply by dinger1986** [reply]
I wouldn't expect it to show up in tactical by installing in mesh.

Just trying to figure out why it wouldn't work with the trmm agent install

**Reply by silversword411** [reply]
You must have a search you're not showing in the TRMM screenshot, unless you have many empty clients configured with no agents anywhere (doesn't make much sense). 

Run server_troubleshooting script https://docs.tacticalrmm.com/troubleshooting/#server-troubleshooting-script

If you clicked quickly on left items without waiting for them to load they will be one click behind what's been selected (All clients for me with 400+ agents take 3-5seconds to load). 

Agent data loads thru port 4222, if mesh is working and agent isn't, it's probably a port 4222 tcp traffic problem

**Reply by dinger1986** [reply]
Can you share a screenshot of the mesh settings on tactical. Again cover any sensitive info

**Reply by Aghiad90** [reply]
@dinger1986  
![image](https://user-images.githubusercontent.com/23388761/177210978-27cd1553-cd0a-42fb-b2f2-4ac091a6ad09.png)

**Reply by Aghiad90** [reply]
@silversword411  i have empty client , this is new setup and there is no agent at all my main problem is that the agent installation will stuck on downloading mesh agent !
i did try manually downloading and installing mesh from mesh.mydomain.com and that is how it did show up on mesh side

**Reply by dinger1986** [reply]
That makes no sense, I don't know why it's not working

**Reply by silversword411** [reply]
Output of server troubleshooting script please

**Reply by Aghiad90** [reply]
Hi @silversword411 
![image](https://user-images.githubusercontent.com/23388761/177216443-999bccf4-ff67-436d-8eb5-c7cb98813930.png)
![image](https://user-images.githubusercontent.com/23388761/177216477-5be6a3ac-ce17-411b-a424-b747f85b354a.png)

and the content of Checklog.log is 

![image](https://user-images.githubusercontent.com/23388761/177216538-13953e95-9a81-40b7-985a-fbf9dac94b0b.png)

**Reply by Aghiad90** [reply]
i think i know why what it is the issue : 
we have the main WAN IP 230 which is the router and from the the router we forward all the traffic from ip 229 to a local server ip 
Tactical RMM detect that the WAN ip is 230 not 229 then block all the connection 

i think it is the way our network been setup create those issue

**Reply by silversword411** [reply]
So your DNS is bad. Fix DNS from agent to server lookup

**Reply by Aghiad90** [reply]
Hi @silversword411  , @dinger1986  
so i did fix the ports issue this is the result of the troubleshoot script 
- if you can see all the ports is open and green 
- i am gussing the reason why i have a red at the beggining is because we have the server behind the firewall and it is not public IP 
- if you ping all the domain it will reply with right ip so no more dns or port issue 

but i still have same problem is there any more ideas that i can try ?
Regards , 

![image](https://user-images.githubusercontent.com/23388761/177429629-a1b32100-5a0a-489f-8e4e-91a8b628f487.png)

**Reply by silversword411** [reply]
The red means how the agent finds the server isn't setup right.

https://docs.tacticalrmm.com/troubleshooting/#from-the-agent

If you'd like paid support send an email to help@amidaware.com
